tests/entities/test_trace_data.py

```diff
@@ -0,0 +1,117 @@
+import mlflow
+from mlflow.entities import SpanType
+from mlflow.entities.span_event import SpanEvent
+
+from tests.tracing.conftest import mock_client as mock_trace_client  # noqa: F401
+
+
+def test_json_deserialization(mock_trace_client):
+    class TestModel:
+        @mlflow.trace()
+        def predict(self, x, y):
+            z = x + y
+
+            with mlflow.start_span(name="with_ok_event") as span:
+                span.add_event(SpanEvent(name="ok_event", attributes={"foo": "bar"}))
+
+            self.always_fail()
+            return z
+
+        @mlflow.trace(span_type=SpanType.LLM, name="always_fail_name", attributes={"delta": 1})
+        def always_fail(self):
+            raise Exception("Error!")
+
+    model = TestModel()
+    try:
+        model.predict(2, 5)
+    except Exception:
+        pass
```

Comment:
```suggestion
    with pytest.raises():
        model.predict(2, 5)
```

can we use `pytest.raises` here?

Reply:
we can, but I think the method is for validating an exception, which is not the intention here?